src/test/java/net/wasdev/wlp/common/plugins/util/InstallFeatureUtilTest.java

```diff
@@ -0,0 +1,189 @@
+    public void setupInstallDir() throws IOException {
+        installDir = temp.newFolder();
+        File src = new File(RESOURCES_INSTALL_DIR);
+        FileUtils.copyDirectory(src, installDir);
+    }
+    
+    @Test
+    public void testConstructor() throws Exception {
+        InstallFeatureUtil util = new InstallFeatureTestUtil(installDir, null, null, new HashSet<String>());
+        assertNotNull(util);
+    }
+    
+    @Test(expected = PluginExecutionException.class)
+    public void testConstructorNoProperties() throws Exception {
+        File olProps = new File(installDir, "lib/versions/openliberty.properties");
+        File wlProps = new File(installDir, "lib/versions/WebSphereApplicationServer.properties");
+        assertTrue(olProps.delete());
+        assertTrue(wlProps.delete());
+        new InstallFeatureTestUtil(installDir, null, null, new HashSet<String>());
+    }
+    
+    @Test
+    public void testConstructorTo() throws Exception {
+        InstallFeatureUtil util = new InstallFeatureTestUtil(installDir, null, "myextension", new HashSet<String>());
+        assertNotNull(util);
+    }
+    
+    /**
+     * TODO remove the expected exception when "from" scenario is supported
+     */
+    @Test(expected = PluginScenarioException.class)
+    public void testConstructorFrom() throws Exception {
+        new InstallFeatureTestUtil(installDir, installDir.getAbsolutePath(), null, new HashSet<String>());
+    }
+    
+    /**
+     * TODO remove the expected exception when installing from ESAs is supported
+     */
+    @Test(expected = PluginScenarioException.class)
+    public void testConstructorEsas() throws Exception {
+        Set<String> esas = new HashSet<String>();
+        esas.add("abc.esa");
+        new InstallFeatureTestUtil(installDir, null, null, esas);
+    }
+    
+    /**
+     * The installFeatures method should be tested from the actual project that
+     * uses it. It will throw an exception here because the test install map jar
+     * does not contain anything.
+     */
+    @Test(expected = PluginExecutionException.class)
+    public void testInstallFeatures() throws Exception {
+        InstallFeatureUtil util = new InstallFeatureTestUtil(installDir, null, null, new HashSet<String>());
+        List<String> featuresToInstall = new ArrayList<String>();
+        featuresToInstall.add("a-1.0");
+        util.installFeatures(true, featuresToInstall);
+    }
+    
+    @Test
+    public void testCombineToSet() throws Exception {
+        Set<String> a = new HashSet<String>();
+        a.add("1");
+        a.add("2");
+        List<String> b = new ArrayList<String>();
+        b.add("1");
+        b.add("3");
+        List<String> c = new ArrayList<String>();
+        b.add("4");
+        b.add("5");
+        Set<String> result = InstallFeatureUtil.combineToSet(a, b, c);
+        assertEquals(5, result.size());
+    }
+    
+    @Test
+    public void testProductInfoValidate() throws Exception {
```

Comment:
What is the purpose of this test? I think that this should really be an integration test. Also, where are there these files? 

- src/test/resources/installdir/bin/productInfo
- src/test/resources/installdir/bin/productInfo.bat
- src/test/resources/installdir/lib/com.ibm.ws.install.map_1.0.21.jar

They are basically empty. Are they used for this test? If so I think they should be deleted as well.

Reply:
Removed this test, will add it as an integration test afterwards in maven/gradle.

Removed these files except the map jar (since the InstallFeatureUtil constructor (invoked by other tests) checks for the presence of that file).